subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/cntr.c

```diff
@@ -5,27 +5,92 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#include <hal/nrf_rtc.h>
+#include <stdint.h>
 
 #include "hal/cntr.h"
 
 #include "hal/debug.h"
 
+#if !defined(CONFIG_BT_CTLR_NRF_GRTC)
+#include <hal/nrf_rtc.h>
 #ifndef NRF_RTC
 #if defined(CONFIG_SOC_COMPATIBLE_NRF54LX)
 #define NRF_RTC NRF_RTC10
 #else /* !CONFIG_SOC_COMPATIBLE_NRF54LX */
 #define NRF_RTC NRF_RTC0
 #endif /* !CONFIG_SOC_COMPATIBLE_NRF54LX */
 #endif /* !NRF_RTC */
+#else /* CONFIG_BT_CTLR_NRF_GRTC */
+#include <hal/nrf_grtc.h>
+#endif /* CONFIG_BT_CTLR_NRF_GRTC */
 
 static uint8_t _refcount;
 
 void cntr_init(void)
 {
+#if defined(CONFIG_BT_CTLR_NRF_GRTC)
+#if defined(CONFIG_BT_CTLR_NRF_GRTC_START)
+	NRF_GRTC->MODE = (GRTC_MODE_SYSCOUNTEREN_Disabled <<
+			  GRTC_MODE_SYSCOUNTEREN_Pos) &
+			 GRTC_MODE_SYSCOUNTEREN_Msk;
+
+	nrf_grtc_task_trigger(NRF_GRTC, NRF_GRTC_TASK_CLEAR);
+
+#if defined(CONFIG_BT_CTLR_NRF_GRTC_KEEPRUNNING)
+	NRF_GRTC->KEEPRUNNING =
+		(GRTC_KEEPRUNNING_REQUEST1_Active <<
+		 GRTC_KEEPRUNNING_REQUEST1_Pos) &
+		GRTC_KEEPRUNNING_REQUEST1_Msk;
+	NRF_GRTC->TIMEOUT = 0U;
+	NRF_GRTC->INTERVAL = 0U;
+	NRF_GRTC->WAKETIME = 4U;
+#endif /* CONFIG_BT_CTLR_NRF_GRTC_KEEPRUNNING */
+
+	NRF_GRTC->CLKCFG = ((GRTC_CLKCFG_CLKSEL_LFXO <<
+			     GRTC_CLKCFG_CLKSEL_Pos) &
+			    GRTC_CLKCFG_CLKSEL_Msk) |
+			   ((GRTC_CLKCFG_CLKFASTDIV_Min <<
+			     GRTC_CLKCFG_CLKFASTDIV_Pos) &
+			    GRTC_CLKCFG_CLKFASTDIV_Msk);
+#endif /* CONFIG_BT_CTLR_NRF_GRTC_START */
+
+	nrf_grtc_event_clear(NRF_GRTC, HAL_CNTR_GRTC_EVENT_COMPARE_TICKER);
+
+	/* FIXME: Replace with nrf_grtc_int_enable when is available,
+	 *        with ability to select/set IRQ group.
+	 *        Shared interrupts is an option? It may add ISR latencies?
+	 */
+	NRF_GRTC->INTENSET1 = HAL_CNTR_GRTC_INTENSET_COMPARE_TICKER_Msk;
```

Comment:
How come you want to explicitly use this interrupt instead of whichever the HAL has allocated for this processor and secure/nonsecure mode?
(note that the accesses thru the hal to SYSCOUNTER will be thru the "domain" the HAL allocated for this processor)